crates/services/p2p/src/service.rs

```diff
@@ -437,6 +437,20 @@ where
         let response_channel = self.request_sender.clone();
         match request_message {
             RequestMessage::Transactions(range) => {
+                let max_len = self
+                    .max_blocks_per_request
+                    .try_into()
+                    .expect("u32 should always fit into usize");
+                if range.len() > max_len {
+                    tracing::error!("Requested range of blocks is too big. Requested length: {:?}, Max length: {:?}", range.len(), max_len);
+                    let response = None;
+                    let _ = self.p2p_service.send_response_msg(
+                        request_id,
+                        ResponseMessage::Transactions(response),
+                    );
+                    return Ok(())
+                }
+
```

Comment:
Nit: This is a very long match arm. While it was pretty long already before this PR, I'd find the code easier to read if we could break this out to a separate function. If we use `tracing::instrument` on that function we could have a cleaner error message, where the lengths are automatically added to the span.

Reply:
https://github.com/FuelLabs/fuel-core/issues/2121

Reply:
Thanks for opening!